frontend/src/app/api/events/route.ts

```diff
@@ -1,20 +1,25 @@
-import { fetchEvents } from '@/lib/data'
 import { NextRequest, NextResponse } from 'next/server'
+import { fetchEventsInWeek } from '@/lib/data'
 
 export async function GET(req: NextRequest) {
   const { searchParams } = new URL(req.url)
-  const limit = searchParams.get('limit')
-  const idCursor = searchParams.get('idCursor')
+  const weekDate = searchParams.get('weekDate')
   const category = searchParams.get('category')
+
+  if (!weekDate) {
+    return NextResponse.json({ error: 'Missing weekDate parameter.' }, { status: 400 })
+  }
+
   try {
-    const events = await fetchEvents(
-      limit ? Number(limit) : 10,
-      idCursor ? Number(idCursor) : undefined,
-      category ? String(category) : undefined,
-    )
+    const week = new Date(weekDate)
+    if (isNaN(week.getTime())) {
+      return NextResponse.json({ error: 'Invalid weekDate parameter.' }, { status: 400 })
```

Comment:
Can we make this more detailed? Is this a failure because week doesn't have a specified time?